### ISSUE TITLE
mistal-chat cude out of memory error

### ISSUE THREAD

**Issue by yuvalshachaf** [mistralai/mistral-finetune#14]
I am getting memory error using the model i have trained from the tutorial, any idea? im using A10G GPU

**Comment by danielhanchen** [comment]
Not a Mistral official product, and shameless promotion, but if you're having OOM issues, have a go with Unsloth :) You get 70%+ memory reduction, 2x faster and no accuracy degradation! https://github.com/unslothai/unsloth Mistral v0.3 7b via a free Colab: https://colab.research.google.com/drive/1_yNCks4BTD5zOnjozppphh5GzMFaMKq_?usp=sharing

**Comment by mosh98** [comment]
reducing the sequence length to under 8000 helped me

**Comment by NathanMayPro** [comment]
Lora is not the cause of you're error
When inference:
- Empty model (13.5gb) + self_attention (depending of the size of you're input) +  activation (depending of number of parameters)
When training:
 - same as previous + gradient (depending of number of parameters) + optimizer 
With LoRa during training:
 - the gradient and optimizer are no longer depending of you're original model size but the LoRa size (A,rank), (rank, B)

**Comment by halilergul1** [comment]
> reducing the sequence length to under 8000 helped me



Hello @mosh98, you mean reducing seq length while training or inference?



Any other updates on this issue, I get the same cuda error while trying to get inferences from the fine-tuned model

**Comment by BlahBlah314** [comment]
Hello,

I have the same problem. The inference with Nemo-instruct is OK on an A100. But when finetuned, using the inference method described in the tutorial with model.load_lora, I get an OOM with the same A100. How did you solve it if you indeed solved it ? Maybe merging the adapter with peft before the inference is a solution ?

**Comment by C3po-D2rd2** [comment]
Same issue here, did anyone found a way?

**Comment by Luis-gd** [comment]
> Lora is not the cause of you're error When inference:
> 
> * Empty model (13.5gb) + self_attention (depending of the size of you're input) +  activation (depending of number of parameters)
>   When training:
> * same as previous + gradient (depending of number of parameters) + optimizer
>   With LoRa during training:
> * the gradient and optimizer are no longer depending of you're original model size but the LoRa size (A,rank), (rank, B)

And what's the issue? It happens to me too and I don't know what to do, IDK if it's something related to load_lora function or what

**Comment by NathanMayPro** [comment]
Lora is something that you add to your model so he cannot reduce the memory usage during loading and inference steps.
He can during training because it reduce the number of trainable parameters (instead of training the huge amount of neurons of the original it create a small model that correct the first one). 
In terms of GPU requirements:
        - Load the model < Inference < Training (In terms of GPU Usage)
Basically you need a bigger GPU.
If you want reduce the GPU memory usage look Quantization

**Comment by Luis-gd** [comment]
> Lora is something that you add to your model so he cannot reduce the memory usage during loading and inference steps.
> He can during training because it reduce the number of trainable parameters (instead of training the huge amount of neurons of the original it create a small model that correct the first one). 
> In terms of GPU requirements:
>         - Load the model < Inference < Training (In terms of GPU Usage)
> Basically you need a bigger GPU.
> If you want reduce the GPU memory usage look Quantization

My problem is not the training process. I trained the model and I have my lora.savetensors.

My problem comes when I use mistral-inference to load lora.savetensors. Before I load lora, my GPU VRAM usage is about 14GB, but when I load lora it gives me a VRAM usage error. My GPU has  24GB, it's good enough.

**Comment by Luis-gd** [comment]
> > Lora is something that you add to your model so he cannot reduce the memory usage during loading and inference steps.
> > He can during training because it reduce the number of trainable parameters (instead of training the huge amount of neurons of the original it create a small model that correct the first one). 
> > In terms of GPU requirements:
> >         - Load the model < Inference < Training (In terms of GPU Usage)
> > Basically you need a bigger GPU.
> > If you want reduce the GPU memory usage look Quantization
> 
> My problem is not the training process. I trained the model and I have my lora.savetensors.
> 
> My problem comes when I use mistral-inference to load lora.savetensors. Before I load lora, my GPU VRAM usage is about 14GB, but when I load lora it gives me a VRAM usage error. My GPU has  24GB, it's good enough.

The only conclusion I can get is the problem I'm facing is made by this load lora method

**Comment by juliendenize** [comment]
Hi! We're archiving this repository as it's no longer actively maintained, and therefore this issue is being closed.
